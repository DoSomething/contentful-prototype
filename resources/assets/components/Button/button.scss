.button {
  &.-inline {
    clear: none;
  }

  &.-align-left {
    float: left;
  }

  &.-align-right {
    float: right;
  }
}
