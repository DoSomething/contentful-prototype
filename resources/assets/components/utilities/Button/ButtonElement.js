import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const ButtonElement = ({ className, isDisabled, onClick, text }) => (
  <button
    className={classnames('btn', className)}
    disabled={isDisabled}
    onClick={onClick}
  >
    {text}
  </button>
);

ButtonElement.propTypes = {
  className: PropTypes.string,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
};

ButtonElement.defaultProps = {
  className: null,
  isDisabled: false,
  onClick: null,
};

export default ButtonElement;
