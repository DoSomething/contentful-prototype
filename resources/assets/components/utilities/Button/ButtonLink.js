import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

/**
 * Basic button link (<a>) with minimal styles.
 * Pass classes to style pseudo states like :hover, :focus and :active, etc.
 *
 * @param {Object} props
 */
const ButtonLink = ({ className, href, onClick, text }) => (
  <a href={href} className={classnames('btn', className)} onClick={onClick}>
    {text}
  </a>
);

ButtonLink.propTypes = {
  className: PropTypes.string,
  href: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
};

ButtonLink.defaultProps = {
  className: null,
  onClick: null,
};

export default ButtonLink;
