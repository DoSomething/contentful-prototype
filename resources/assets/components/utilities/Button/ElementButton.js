/* eslint-disable react/button-has-type */

import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

/**
 * Basic button element (<button>) with minimal styles.
 * Pass classes to style pseudo states like :hover, :focus and :active, etc.
 *
 * @param {Object} props
 */
const ElementButton = ({
  className,
  data,
  isDisabled,
  onClick,
  text,
  type,
}) => {
  return (
    <button
      className={classnames('btn', className)}
      disabled={isDisabled}
      onClick={onClick}
      type={type}
      {...data}
    >
      {text}
    </button>
  );
};

ElementButton.propTypes = {
  className: PropTypes.string,
  data: PropTypes.object,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
};

ElementButton.defaultProps = {
  className: null,
  data: {},
  isDisabled: false,
  onClick: null,
  type: 'button',
};

export default ElementButton;
