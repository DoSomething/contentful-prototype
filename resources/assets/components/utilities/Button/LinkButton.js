import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { isExternal } from '../../../helpers/url';

/**
 * Basic button link (<a>) with minimal styles.
 * Pass classes to style pseudo states like :hover, :focus and :active, etc.
 *
 * @param {Object} props
 */
const LinkButton = ({ attributes, className, href, onClick, text }) => {
  const linkAttributes = {
    rel: isExternal(href) ? 'noopener noreferrer' : null,
    target: isExternal(href) ? '_blank' : null,
  };

  return (
    <a
      className={classnames('btn', className)}
      href={href}
      onClick={onClick}
      {...linkAttributes}
      {...attributes}
    >
      {text}
    </a>
  );
};

LinkButton.propTypes = {
  attributes: PropTypes.object,
  className: PropTypes.string,
  href: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
};

LinkButton.defaultProps = {
  attributes: {},
  className: null,
  onClick: null,
};

export default LinkButton;
