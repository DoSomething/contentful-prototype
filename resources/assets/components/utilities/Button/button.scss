@import '../../../scss/next-toolbox.scss';

.button {
  text-transform: capitalize;

  &.-inline {
    clear: none;
  }

  &.-align-left {
    float: left;
  }

  &.-align-right {
    float: right;
  }

  &.-secondary,
  &.is-loading {
    &:focus {
      border: 0 none;
      box-shadow: none;
    }
  }
}

.button.-attached {
  width: 100%;
}

.button__attached-container {
  padding: theme('spacing.3');
}
