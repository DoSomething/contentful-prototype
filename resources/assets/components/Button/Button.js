import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './button.scss';

const Button = ({ onClick, classNames, text }) => {
  const handleOnClick = () => {
    onClick();
  };

  return (
    <button className={classnames('button', classNames)} onClick={handleOnClick}>{text}</button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  classNames: PropTypes.string,
  text: PropTypes.string,
};

Button.defaultProps = {
  classNames: null,
  text: 'Join Us',
};

export default Button;
