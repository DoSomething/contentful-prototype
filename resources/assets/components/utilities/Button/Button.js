import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import requiredIf from 'react-required-if';

import './button.scss';

const Button = ({
  onClick,
  className,
  type,
  attached,
  disabled,
  loading,
  children,
}) => {
  const classNames = classnames(
    'button',
    {
      'is-loading': loading,
      'is-disabled': disabled,
      '-attached': attached,
    },
    className,
  );

  const buttonText = Array.isArray(children) ? children.join('') : children;
  const button = (
    <button
      type={type}
      className={classNames}
      disabled={disabled || loading}
      onClick={onClick}
    >
      {buttonText}
    </button>
  );

  // If we're rendering an "attached" button, wrap it in a padded container:
  if (attached) {
    return <div className="button__attached-container">{button}</div>;
  }

  return button;
};

Button.propTypes = {
  type: PropTypes.string,
  onClick: requiredIf(PropTypes.func, props => props.type !== 'submit'),
  className: PropTypes.string,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  attached: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
};

Button.defaultProps = {
  type: null,
  className: null,
  disabled: false,
  loading: false,
  attached: false,
};

export default Button;
