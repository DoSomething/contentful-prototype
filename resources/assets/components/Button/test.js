import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('Button snapshot test', () => {
  const tree = renderer.create(
    <Button callback={() => {}} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Decked out Button test', () => {
  const tree = renderer.create(
    <Button classNames="-modifier" callback={() => {}} callbackArgument="1234" callbackMetadata={{ source: 'jest test' }} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
