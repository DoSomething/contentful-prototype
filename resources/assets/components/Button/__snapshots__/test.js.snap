// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot test 1`] = `
<button
  className="button -modifier"
  disabled={false}
  onClick={[Function]}
>
  Join Us
</button>
`;
